### PR TITLE
defaults for saved query name and description

### PR DESCRIPTION
**Pull request by jrchudy**
This PR adds default values into the name and description columns during saved query creation. These values will populate the form when it is opened and can be replaced by other values by the user.

This also fixes an issue with markdown inputs not displaying correctly in the saved query popup that hasn't been encountered yet.

### CODE REVIEW

**Review by RFSH — CHANGES_REQUESTED** [review]
If our goal here is to use what users see in the chiclets, then you shouldn't use the raw stable facet. The stable facet is going to give you the "raw values", so it won't apply any formatting and in case of entity modes it will be the raw value instead of rowname.

You should do similar things that we did for chiclets. You can use `vm.reference.facetColumns[i].displayname` to get the displayname of a facet, and `vm.facetModels[i].appliedFilters` to get the list of filters that are applied to that facet. You can see how this is done [here](https://github.com/informatics-isi-edu/chaise/blob/master/common/templates/recordset.html#L94,L118)

File: common/table.js [hunk 1]
```diff
@@ -1527,10 +1527,85 @@
...
+                    var description = nameDescriptionPrefix + ":\n";
+
+                    var modelsWFilters = scope.vm.facetModels.filter(function (fm, idx) {
+                        fm.displayname = scope.vm.reference.facetColumns[idx].displayname.value;
+                        fm.preferredMode = scope.vm.reference.facetColumns[idx].preferredMode;
+                        return (fm.appliedFilters.length > 0);
+                    });
+
+                    if (scope.vm.search) {
+                        name += " " + scope.vm.search;
```
Comment (RFSH) [inline common/table.js:1573]: I think you need a separator between search and the rest.

File: common/table.js [hunk 2]
```diff
@@ -1527,10 +1527,85 @@
...
+                            if (name === null) {
+                                str += " _No value_";
+                            } else if (name === "<i>All records with value </i>") {
+                                str += " _All records with value_"
+                            } else if (name === '') {
+                                str += " _Empty_"
+                            } else {
+                                str += " " + name;
+                            }
+                            if (idx+1 != options.length) str += ", "
```
Comment (RFSH) [inline common/table.js:1550]: This `", "` is causing two whitespaces after comma, is that intentional?

File: common/table.js [hunk 3]
```diff
@@ -1527,10 +1527,85 @@
...
+                        });
+                        return str;
+                    }
+
+                    // thresholds for when to use a modified simpler syntax for the name
+                    var facetChoicesThreshold = 5; // default to 5 choices
+                    var facetTextLengthThreshold = 60; // default to 50 characters;
+                    var nameLengthThreshold = 200; // default to 200 characters
+
+                    var nameDescriptionPrefix = scope.vm.reference.table.name + " with";
```
[the remark this answers is not in the document]
Comment (jrchudy) [reply common/table.js:1560]: fixed

File: common/table.js (hunk 1 above)
Comment (jrchudy) [reply common/table.js:1573]: fixed

**Comment by jrchudy** [comment]
I added chaise config properties for the thresholds, added an explanation in the code for the default name/description, and added chaise config documentation. We can decide on the chaise config property names with hongsuda in chaise meeting.

**Comment by jrchudy** [comment]
Closing this pull request since #2134 also includes changes for saved query functionality and is based off of this branch.

**Pull request closed without merging**